### PR TITLE
Ft user records

### LINKED ISSUE #4: [FEATURE] Create User Record
### Implement the following User Record related tasks:
1. User Record model, validations, and tests.
2. User Record Controller and tests.
3. Code linting.

### PR DESCRIPTION
**Pull request by katungi**
## Description
Added the ability for the users to add their records. It involves addition of a controller and model.

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #(issue:#4) 

## How Has This Been Tested?
I wrote model and request tests and the code passed successfully.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/controllers/records_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    def destroy
+        @record.destroy 
+        head :no_content
+    end
+
+    private 
+
+    def record_params 
+        params.permit( :user_id, :specialist_id, :description, :diagnosis)
```
Comment (wathigo) [inline app/controllers/records_controller.rb]: Lets use `params.require(:record).permit( :user_id, :specialist_id, :description, :diagnosis)`

**Review by wathigo — CHANGES_REQUESTED** [review]
Great work @katungi 
Check my comment :point_down:

**Review by wathigo — APPROVED** [review]
LGTM!

File: app/controllers/records_controller.rb (hunk 1 above)
Comment (katungi) [reply app/controllers/records_controller.rb]: thanks for that @wathigo, worked on it.

**Review by Danielshow — APPROVED** [review]
LGTM!

**Pull request merged**